File: Tellma/ClientApp/src/app/data/dto/generate-markup-arguments.ts
export interface GenerateMarkupArguments {
    culture?: string;
}

export interface GenerateMarkupByFilterArguments extends GenerateMarkupArguments {
    filter?: string;
    orderby?: string;
    top?: number;
    skip?: number;
    i?: (string | number)[];
}

// tslint:disable-next-line:no-empty-interface
export interface GenerateMarkupByIdArguments extends GenerateMarkupArguments {
    // id?: string | number;
}

File: Tellma/Controllers/DTO/GenerateMarkupArguments.cs
using System.Collections.Generic;

namespace Tellma.Controllers.Dto
{
    public class GenerateMarkupArguments
    {
        public string Culture { get; set; }
    }

    public class GenerateMarkupByFilterArguments<TKey> : GenerateMarkupArguments
    {
        public string Filter { get; set; }
        public string OrderBy { get; set; }
        public int Top { get; set; }
        public int Skip { get; set; }
        public List<TKey> I { get; set; }
    }

    public class GenerateMarkupByIdArguments : GenerateMarkupArguments
    {
      //  public object Id { get; set; }
    }
}
